### PR TITLE
FIX: interpret analog input values as int16

### PR DESCRIPTION
**Pull request by JJL772**
Both signed and unsigned representations behave like a signed 16-bit integer. In the case of unsigned, the value is limited to the range 0-32767, the former is limited to -32768-32767. As such, treating the values from the device as int16 (and thus sign extending it when assigning to RVAL) is the correct way of doing things.

MSB representation is still not handled in this PR.

Leaving as draft until I can find and test with a bipolar terminal!

### CODE REVIEW

**Comment by JJL772** [comment]
## EL331X

Signed representation and absolute value w/MSB sign are the only two representations mentioned in the docs.

image

## EL30XX, EL31XX

This covers bipolar terminals like EL3174, too.

Signed, unsigned and absolute w/MSB sign are supported.

Ranges specified in docs:
- signed: -32768 to 32767 
image

- unsigned: 0 to 32767 
image

- absolute MSB sign: 0 to 32767, sign 0 or 1, so -32767 to 326767 total 
image


Absolute w/MSB sign seems to only exist to provide a symmetrical range

tldr; this is safe to merge after testing :)

**Comment by JJL772** [comment]
Closing because I renamed the branch.
